File: SiHyeon/week15/정수 삼각형.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> triangle)
{
    int answer = 0;
    // 밑에서부터 위에 더했을때 큰 값을 더해줌
    for (int i = triangle.size() - 1; i > 0; i--)
    {
        for (int j = 0; j < triangle[i].size() - 1; j++)
        {
            if (triangle[i][j] > triangle[i][j + 1])
                triangle[i - 1][j] += triangle[i][j];
            else
                triangle[i - 1][j] += triangle[i][j + 1];
        }
    }
    answer = triangle[0][0];
    return answer;
}

File: eunbi/week15/정수삼각형.java
package algorithmStudy;

import java.util.*;

class Solution {
    public int solution(int[][] triangle) {
        int answer = 0;
        
        for(int i=1; i<triangle.length; i++){
            for(int j=0; j<triangle[i].length; j++){
                if(j==0)
                    triangle[i][j] += triangle[i-1][j];
                else if(j==triangle[i].length-1)
                    triangle[i][j] += triangle[i-1][j-1];
                else
                    triangle[i][j] += Math.max(triangle[i-1][j], triangle[i-1][j-1]);
            }
        }
        
        for(int i : triangle[triangle.length-1])
            answer = Math.max(answer, i);
        
        return answer;
    }
}
